#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("AutoExp",     bool_t,   0, "自动曝光",         False)
gen.add("ExpTime",     int_t,    0, "手动曝光时间",      3000,  500, 5000)
gen.add("Resolution",  str_t,    0, "分辨率模式",       "MaxSize")
gen.add("Gain",        double_t, 0, "图像增益",         100, 10, 300)
gen.add("AutoWB",      bool_t,   0, "自动白平衡",       False)
gen.add("CameraName",  str_t,    0, "相机名称",         "")
gen.add("SetOnceWB",   bool_t,   0, "手动设置一次白平衡", False)
gen.add("Save",        bool_t,   0, "保存参数",         False)

exit(gen.generate("md_camera", "md_camera_node", "Camera"))
